comp/forwarder/defaultforwarder/forwarder_health.go

```diff
@@ -206,14 +246,23 @@ func (fh *forwarderHealth) validateAPIKey(apiKey, domain string) (bool, error) {
 	}
 
 	fh.setAPIKeyStatus(apiKey, domain, &apiKeyUnexpectedStatusCode)
-	return false, fmt.Errorf("Unexpected response code from the apikey validation endpoint: %v", resp.StatusCode)
+	return false, fmt.Errorf("unexpected response code from the apikey validation endpoint: %v", resp.StatusCode)
 }
 
-func (fh *forwarderHealth) hasValidAPIKey() bool {
+// check if any of the endpoints have a valid apiKey, updating health state
+func (fh *forwarderHealth) checkValidAPIKey() bool {
 	validKey := false
 	apiError := false
 
+	// mutex just to copy the map, to avoid holding onto it for too long
+	fh.keyMapMutex.Lock()
+	keysPerDomain := make(map[string][]string)
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏tiniest nitpick ever: you can move the `make` outside the `Lock`